collect_app/src/main/java/org/odk/collect/android/tasks/InstanceUploaderTask.java

```diff
@@ -490,8 +496,14 @@ private boolean uploadOneSubmission(String urlString, String id, String instance
                         outcome.mResults.put(id, fail + response.getStatusLine().getReasonPhrase()
                                 + " (" + responseCode + ") at " + urlString);
                     } else {
-                        outcome.mResults.put(id, fail + response.getStatusLine().getReasonPhrase()
-                                + " (" + responseCode + ") at " + urlString);
+                        // If response from server is valid use that else use default messaging
+                        if (messageParser != null && messageParser.isValid()){
```

Comment:
Humm, I did not have them turned on. Thanks for the schooling, im always trying to get better! Should be fixed now.